### ISSUE TITLE
[Tooltip] Keep tooltip open when trigger is clicked

### ISSUE THREAD

**Issue by giubatt** [radix-ui/primitives#2029]
### Current Behavior

`onPointerDownOutside={(e) => e.preventDefault(); }` on `Tooltip.Content` is not enough on it's own to prevent clicks outside the tooltip from closing it, you also need to `event.preventDefault` on the `Tooltip.Trigger` `onClick` event. 

### Expected behavior

`onPointerDownOutside` should be enough to provide this behavior, as preventing default on the `Trigger` click event may have unintended consequences because of preventing the event from propagating

### Reproducible example

[CodeSandbox Example](https://codesandbox.io/p/sandbox/interesting-gagarin-q92fus?file=%2FApp.jsx&selection=%5B%7B%22endColumn%22%3A20%2C%22endLineNumber%22%3A13%2C%22startColumn%22%3A20%2C%22startLineNumber%22%3A13%7D%5D)

### Additional context

I found this bug after changing my `Tooltip` to have both `preventDefaults`, and noticing that when using it alongside radix `ToggleGroup` it was stopping ToggleGroup from working. This was basically my setup with both of them:
```tsx
<ToggleGroup.Root
  type="single"
  value={view}
  onValueChange={(value: "list" | "map" | "scenarios") => {
    if (value) router.replace({ query: { ...router.query, tab: value } })
  }}
>
  <Tooltip content="Show List">
    <ToggleGroup.Item value="list" aria-label="List">
      <List /> {/* icon */}
    </ToggleGroup.Item>
  </Tooltip>
  <Tooltip content="Show Map">
    <ToggleGroup.Item value="map" aria-label="Map">
      <MapTrifold /> {/* icon */}
    </ToggleGroup.Item>
  </Tooltip>
  <Tooltip content="Show Other stuff">
    <ToggleGroup.Item value="scenarios" aria-label="Other Stuff">
      <Square /> {/* icon */}
    </ToggleGroup.Item>
  </Tooltip>
</ToggleGroup.Root>
```
Having the Tooltip.Trigger with click event.preventDefault() would stop the even from bubbling up to ToggleGroup.Root (my guess?) which would make it stop working/toggling 

### Your environment

Same as the codesandbox, forked from radix tooltip docs

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |         |         |
| React            | n/a     |         |
| Browser          |         |         |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         |         |         |
| Operating System |         |         |

**Comment by benoitgrelard** [comment]
Hey @giubatt, unfortunately that is kinda of a side-effect of the implementation.
Tooltips generally should close on activation (be it pointer or keyboard) so we use `onClick` for that.
The issue is really that `event.preventDefault` in the platform is loaded and often has unintended side-effects.
It would be great to have more control (see [this tweet](https://twitter.com/devongovett/status/1633962215114956801)) but unfortunately it doesn't exist.

I will close this as I don't think we can do anything about it, and this is definitely what we would consider an edge case (keeping it open on activation).

**Comment by giubatt** [comment]
Yeah, I understand. I'm not sure what could be done about it, only thing I can think of is use a context to trigger the onValueChange without event propagation, but I'm not sure what the consequences of this would be in the current implementation.

It's not a blocker for me currently anyway, the only use case I had for this (and I'm just preventing default when using the Tooltip, which is fine) was for a CopyToClipboard component, so that the text in the tooltip changes without it closing and reopening (for feedback that the text as copied)

**Comment by vincerubinetti** [comment]
Here's something I'm doing. It's for a little help (?) icon next to fields. I want the tooltip to open when focusing or hovering the (?), but also when clicking, for users (including me) who reflexively click it. I'm not sure about the consequences of this, but I feel like clicking to show the tooltip is also good for accessibility.

```tsx
<button
  onClick={async (event) => {
    const target = event.currentTarget;
    if (!target) return;
    await sleep(); // basically a setTimeout of 0
    target.blur();
    target.focus();
    event.stopPropagation();
  }}
/>
```

**Comment by hafijAsadMVN** [comment]
> This is achievable with the current API.
> 
> We have an example in our storybook: https://github.com/radix-ui/primitives/blob/main/packages/react/tooltip/src/Tooltip.stories.tsx#L659-L687
> 
> I will add an example in the docs as this question comes up once in a while.

I've tried with your example, its not working:

```
         <TooltipProvider>
                <Tooltip
                    delayDuration={200}
                    onOpenChange={(status) => {
                        setOpen(status)
                        if (!status) {
                            setParentCat({} as CategoryNavigationContentModel);
                            setChildCat({} as CategoryNavigationItemModel)
                        }
                    }}>
                    <TooltipTrigger
                        onClick={(event) => event.preventDefault()}
                        ref={triggerRef}
                    >
                        <Button
                            variant={"outline"}
                            className={`tw-gap-2 !tw-p-0 lg:!tw-px-4 tw-h-[26px] tw-w-[26px] md:tw-h-[36px] md:tw-w-[36px] lg:tw-h-[40px] lg:tw-w-fit ${open ? "tw-border-mvnGreen-800 tw-bg-mvnGreen-10" : ""}`}
                        >
                            <HamBurgerIcon/>
                            <span className="tw-hidden lg:tw-block">Categories</span>
                            <ChevronDown width={16} className="tw-hidden lg:tw-block"/>
                        </Button>
                    </TooltipTrigger>
                    <TooltipContent
                        side={"bottom"}
                        align={"start"}
                        sideOffset={2}
                        className={"tw-w-full !tw-p-0"}
                        onPointerDownOutside={(event) => {
                            if (event.target === triggerRef.current) event.preventDefault();
                        }}
                    >
                    ......
                    ```

**Comment by olgabast** [comment]
@hafijAsadMVN it works for me with this change:
```
onPointerDownOutside={(event) => {
  if (triggerRef.current?.contains(event.target)) event.preventDefault()
}}
```

**Comment by youneszahzouh** [comment]
none of the above worked for me , until i removed the check of ref ( if (triggerRef.current?.contains(event.target)) ), here is what's working for me:
```js

<TooltipProvider>
        <Tooltip delayDuration={0}>
          <TooltipTrigger
            onClick={(event) => {
              event.preventDefault();
            }}
          >
            <MdOutlineContentCopy
              onClick={() => {
                navigator.clipboard.writeText("url || translate()");
                setCopyTooltipText("Copied!");
              }}
              size={36}
              className="hidden cursor-pointer text-secondary-color hover:drop-shadow-lg md:block"
            />
          </TooltipTrigger>
          <TooltipContent
            onPointerDownOutside={(event) => {
              event.preventDefault();
            }}
          >
            {copyTooltipText}
          </TooltipContent>
        </Tooltip>
      </TooltipProvider>


```

**Comment by kerryj89** [comment]
Are these solutions supposed to close the Tooltip and re-open when clicking on the trigger?  It's far better than it was, but just wondering if this is as good as we could get it.  Ideally the tooltip won't animate.

 
https://github.com/user-attachments/assets/4dbc48e4-5f47-4984-8f87-b42c96e4f89f

**Comment by DavidNepozitek** [comment]
https://github.com/radix-ui/primitives/pull/3380 introduced closing the tooltip when `onPointerDown` event is triggered.
Thus, the previous solutions stopped working. To fix that, it's necessary to add `onPointerDown={(event) => (event.preventDefault())}` to the trigger, as well.

Full minimal example:
```
<TooltipProvider>
    <Tooltip>
        <TooltipTrigger
            onClick={(event) => event.preventDefault()}
            onPointerDown={(event) => event.preventDefault()}
        >
            Trigger Text
        </TooltipTrigger>
        <TooltipContent onPointerDownOutside={(event) => {event.preventDefault();}}>
            {tooltipText}
        </TooltipContent>
    </Tooltip>
</TooltipProvider>
```

**Comment by yelsayed** [comment]
@DavidNepozitek thank you for this!

**Comment by arsamurai** [comment]
This helped me
But now the trigger don't work on hover

If you need set "input" as trigger

const Tooltip = ({ children, content, ...props }) => {
  const [open, setOpen] = useState(true)

  return (
    <TooltipProvider>
      <TooltipRoot open={open} onOpenChange={() => setOpen(false)}>
        <TooltipTrigger onClick={() => setOpen(true)} asChild>
          {children}
        </TooltipTrigger>
        <TooltipContent side="bottom" sideOffset={8} align="start" {...props}>
          {content}
        </TooltipContent>
      </TooltipRoot>
    </TooltipProvider>
  )
}
